pkg/registry/resource/podscheduling/storage/storage.go

```diff
@@ -0,0 +1,138 @@
+	"k8s.io/apiserver/pkg/registry/generic"
+	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/kubernetes/pkg/apis/resource"
+	"k8s.io/kubernetes/pkg/printers"
+	printersinternal "k8s.io/kubernetes/pkg/printers/internalversion"
+	printerstorage "k8s.io/kubernetes/pkg/printers/storage"
+	"k8s.io/kubernetes/pkg/registry/resource/podscheduling"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+)
+
+// REST implements a RESTStorage for PodSchedulings.
+type REST struct {
+	*genericregistry.Store
+}
+
+// NewREST returns a RESTStorage object that will work against PodSchedulings.
+func NewREST(optsGetter generic.RESTOptionsGetter) (*REST, *StatusREST, error) {
+	store := &genericregistry.Store{
+		NewFunc:                  func() runtime.Object { return &resource.PodScheduling{} },
+		NewListFunc:              func() runtime.Object { return &resource.PodSchedulingList{} },
+		PredicateFunc:            podscheduling.Match,
+		DefaultQualifiedResource: resource.Resource("podschedulings"),
+
+		CreateStrategy:      podscheduling.Strategy,
+		UpdateStrategy:      podscheduling.Strategy,
+		DeleteStrategy:      podscheduling.Strategy,
+		ReturnDeletedObject: true,
+		ResetFieldsStrategy: podscheduling.Strategy,
+
+		TableConvertor: printerstorage.TableConvertor{TableGenerator: printers.NewTableGenerator().With(printersinternal.AddHandlers)},
+	}
+	options := &generic.StoreOptions{RESTOptions: optsGetter, AttrFunc: podscheduling.GetAttrs}
+	if err := store.CompleteWithOptions(options); err != nil {
+		return nil, nil, err
+	}
+
+	statusStore := *store
+	statusStore.UpdateStrategy = podscheduling.StatusStrategy
+	statusStore.ResetFieldsStrategy = podscheduling.StatusStrategy
+
+	rest := &REST{store}
+	store.Decorator = rest.defaultOnRead
+
+	return rest, &StatusREST{store: &statusStore}, nil
+}
+
+// defaultOnRead sets interlinked fields that were not previously set on read.
+// We can't do this in the normal defaulting path because that same logic
+// applies on Get, Create, and Update, but we need to distinguish between them.
+//
+// This will be called on both PodScheduling and PodSchedulingList types.
+func (r *REST) defaultOnRead(obj runtime.Object) {
+	switch s := obj.(type) {
+	case *resource.PodScheduling:
+		r.defaultOnReadClaim(s)
+	case *resource.PodSchedulingList:
+		r.defaultOnReadClaimList(s)
+	default:
+		// This was not an object we can default.  This is not an error, as the
+		// caching layer can pass through here, too.
+	}
+}
+
+// defaultOnReadClaimList defaults a PodSchedulingList.
+func (r *REST) defaultOnReadClaimList(list *resource.PodSchedulingList) {
+	if list == nil {
+		return
+	}
+
+	for i := range list.Items {
+		r.defaultOnReadClaim(&list.Items[i])
+	}
+}
+
+// defaultOnRead defaults a single PodScheduling.
+func (r *REST) defaultOnReadClaim(claim *resource.PodScheduling) {
+	if claim == nil {
+		return
+	}
+
+	// TODO: can this and the code above be removed?
```

Comment:
Yes, the default-on-read stuff is pretty rarely needed, and is usueally a bandaid on existing APIs not new ones

Reply:
Removed.